api/src/controllers/young/inscription.js

```diff
@@ -45,4 +46,70 @@ router.put("/profile", passport.authenticate("young", { session: false, failWith
   }
 });
 
+router.put("/availability", passport.authenticate("young", { session: false, failWithError: true }), async (req, res) => {
+  try {
+    const young = await YoungObject.findById(req.user._id);
+    if (!young) return res.status(404).send({ ok: false, code: ERRORS.NOT_FOUND });
+
+    const sessions = await getCohortSessionsAvailability(young);
+
+    const sessionsId = sessions.map((session) => {
+      return session.id;
+    });
+
+    const { error, value } = Joi.object({
+      cohort: Joi.string()
+        .trim()
+        .required()
+        .valid(...sessionsId),
+    }).validate(req.body);
+
+    if (error) {
+      return res.status(400).send({ ok: false, code: ERRORS.INVALID_PARAMS });
+    }
+
+    young.set({ inscriptionStep: STEPS.PARTICULIERES, ...value });
+    await young.save({ fromUser: req.user });
+    await inscriptionCheck(value, young, req);
+
+    return res.status(200).send({ ok: true, data: young });
+  } catch (error) {
+    capture(error);
+    return res.status(500).send({ ok: false, code: ERRORS.SERVER_ERROR, error });
```

Comment:
```suggestion
    return res.status(500).send({ ok: false, code: ERRORS.SERVER_ERROR });
```

Dans une autre tache, on ne fait plus suivre les erreurs côté client quand elles sont innatendues : elles donnent potentiellement des informations